app/src/ui/discard-changes/discard-changes-dialog.tsx

```diff
@@ -84,7 +84,9 @@ export class DiscardChanges extends React.Component<
 
         <DialogFooter>
           <ButtonGroup destructive={true}>
-            <Button type="submit">Cancel</Button>
+            <Button disabled={this.state.isDiscardingChanges} type="submit">
```

Comment:
It looks like you made this change before I got around to explaining how we can do this using the built-in functionality of the `Dialog` component. See [comment].(https://github.com/desktop/desktop/pull/7029#issuecomment-470996180)

I can made the change for you if you like, just let me know 👍

Reply:
Looks this should be how this? When the process started it's making dialog not dismissable. When process ended dialog closes automatically.

```diff
diff --git a/app/src/ui/discard-changes/discard-changes-dialog.tsx b/app/src/ui/discard-changes/discard-changes-dialog.tsx
index 32a901f8e..0a8fba43a 100644
--- a/app/src/ui/discard-changes/discard-changes-dialog.tsx
+++ b/app/src/ui/discard-changes/discard-changes-dialog.tsx
@@ -60,6 +60,7 @@ export class DiscardChanges extends React.Component<
 
   public render() {
     const discardingAllChanges = this.props.discardingAllChanges
+    const isDiscardingChanges = this.state.isDiscardingChanges
 
     return (
       <Dialog
@@ -70,7 +71,8 @@ export class DiscardChanges extends React.Component<
             : toPlatformCase('Confirm Discard Changes')
         }
         onDismissed={this.props.onDismissed}
-        loading={this.state.isDiscardingChanges}
+        dismissable={isDiscardingChanges ? false : true}
+        loading={isDiscardingChanges}
         type="warning"
       >
         <DialogContent>
@@ -84,13 +86,10 @@ export class DiscardChanges extends React.Component<
 
         <DialogFooter>
           <ButtonGroup destructive={true}>
-            <Button disabled={this.state.isDiscardingChanges} type="submit">
+            <Button disabled={isDiscardingChanges} type="submit">
               Cancel
             </Button>
-            <Button
-              onClick={this.discard}
-              disabled={this.state.isDiscardingChanges}
-            >
+            <Button onClick={this.discard} disabled={isDiscardingChanges}>
               {discardingAllChanges
                 ? toPlatformCase('Discard All Changes')
                 : toPlatformCase('Discard Changes')}
```

Reply:
@billygriffin how do you feel about the interaction explained above? Basically, once you start discarding, you can't back out of it (dialog is not dismissable and all buttons are disabled).

Reply:
@HashimotoYT, yes, that looks good!